core/commonMain/src/kotlinx/serialization/modules/SerializersModuleCollector.kt

```diff
@@ -76,12 +78,22 @@ public interface SerializersModuleCollector {
     /**
      * Accept a default deserializer provider, associated with the [baseClass] for polymorphic deserialization.
      *
-     * This will not affect serialization.
+     * This function affect only deserialization process. To avoid confusion, it was deprecated and replaced with [polymorphicDefaultDeserializer].
+     * To affect serialization process, use [SerializersModuleCollector.polymorphicDefaultSerializer].
      *
-     * @see SerializersModuleBuilder.polymorphicDefaultDeserializer
-     * @see PolymorphicModuleBuilder.defaultDeserializer
+     * [defaultDeserializerProvider] is invoked when no polymorphic serializers associated with the `className`
+     * in the scope of [baseClass] were found. `className` could be `null` for formats that support nullable class discriminators
+     * (currently only `Json` with `useArrayPolymorphism` set to `false`).
+     *
+     * [defaultDeserializerProvider] can be stateful and lookup a serializer for the missing type dynamically.
+     *
+     * @see SerializersModuleCollector.polymorphicDefaultDeserializer
+     * @see SerializersModuleCollector.polymorphicDefaultSerializer
      */
-    // TODO: deprecate in 1.4
+    @Deprecated(
+        "Deprecated in favor of function with more precise name: polymorphicDefaultDeserializer",
+        ReplaceWith("polymorphicDefaultDeserializer(baseClass, defaultDeserializerProvider)")
```

Comment:
Please leave an explicit `level` and a comment like `// WARNING since 1.5.0, ERROR since 1.6.0, HIDDEN since 1.7.0`

Reply:
Also, documentation can be `@suppress`-ed now

Reply:
I think we shouldn't delete documentation unless function itself is HIDDEN